pkg/server/server.go

```diff
@@ -163,6 +207,28 @@ func (s *BCDBHTTPServer) serveRequests(l net.Listener) {
 	s.logger.Infof("Finished serving requests on: %s", s.listen.Addr().String())
 }
 
+func (s *BCDBHTTPServer) servePrometheus() {
+	if s.metricsServer == nil {
+		return
+	}
```

Comment:
isn't it better to do this around the the `go s.servePrometheus()`?

Reply:
They are both equally good, IMO. What do you prefer?